wandb/sdk/interface/router_relay.py

```diff
@@ -4,35 +4,44 @@
 
 """
 
+from __future__ import annotations
+
 from typing import TYPE_CHECKING
 
+from wandb.proto import wandb_internal_pb2 as pb
+from wandb.proto import wandb_server_pb2 as spb
 from wandb.sdk.mailbox import Mailbox
 
 from .router_queue import MessageQueueRouter
 
 if TYPE_CHECKING:
     from queue import Queue
 
-    from wandb.proto import wandb_internal_pb2 as pb
-
 
 class MessageRelayRouter(MessageQueueRouter):
-    _relay_queue: "Queue[pb.Result]"
+    _relay_queue: Queue[pb.Result]
 
     def __init__(
         self,
-        request_queue: "Queue[pb.Record]",
-        response_queue: "Queue[pb.Result]",
-        relay_queue: "Queue[pb.Result]",
+        request_queue: Queue[pb.Record],
+        response_queue: Queue[pb.Result],
+        relay_queue: Queue[pb.Result],
         mailbox: Mailbox,
     ) -> None:
         self._relay_queue = relay_queue
         super().__init__(
-            request_queue=request_queue, response_queue=response_queue, mailbox=mailbox
+            request_queue=request_queue,
+            response_queue=response_queue,
+            mailbox=mailbox,
         )
 
-    def _handle_msg_rcv(self, msg: "pb.Result") -> None:
-        if msg.control.relay_id:
-            self._relay_queue.put(msg)
-            return
-        super()._handle_msg_rcv(msg)
+    def _handle_msg_rcv(self, msg: pb.Result | spb.ServerResponse) -> None:
+        if isinstance(msg, pb.Result):
+            relay_msg = msg
+        else:
+            relay_msg = msg.result_communicate
+
+        if relay_msg.control.relay_id:
```

Comment:
i think would be good to add a quick comment about this one...

Reply:
What should it say? If it matters: this is `legacy-service` logic, and AFAIK it never hits the `else` branch, though I'm not sure enough of that to delete it.

Reply:
that would be good enough, i had to go back and forth in the file to understand this, so just a small comment here would be good enough...